manticore/core/smtlib/expression.py

```diff
@@ -663,7 +663,7 @@ def __init__(
         self, index_bits: int, index_max: Optional[int], value_bits: int, *operands, **kwargs
     ):
         assert index_bits in (32, 64, 256)
-        assert value_bits in (8, 16, 32, 64, 256)
+        assert value_bits in (1, 8, 16, 32, 64, 256)
```

Comment:
This is the reason for that change:
https://github.com/trailofbits/manticore/blob/2ff301ebedc960ab975468f7615afb4f888876ee/manticore/platforms/evm_world_state.py#L23-L29
`map` records which (symbolic) offsets have been written to.  `map` is an `ArrayProxy`, which inherits from `Array`.  True, "written to" is a Boolean condition.  But if we had an `ArrayBool`, we would also require an `ArrayBoolProxy` to make this approach work, right?  Is there a better approach?

Reply:
> If we had an ArrayBool, we would also require an ArrayBoolProxy to make this approach work, right?

Makes sense to me. 

> Is there a better approach?

I'm fine with this, but perhaps @feliam will have a stronger opinion?

Reply:
Sorry, I actually meant the requirement for an `ArrayBoolProxy` class to be a negative.  I think it would create unnecessary code duplication.

I guess what I should ask, what would be the "win" from using such `Bool` classes?

Reply:
Apologies, I think my comment was ambiguous. I'm fine with leaving it as-is because I don't see any harm that allowing 1-bit BitVectors would cause, but Felipe might be aware of some hidden problems there.